core/src/main/java/org/bitcoinj/core/AddressParser.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright by the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bitcoinj.core;
+
+import org.bitcoinj.base.BitcoinNetwork;
+import org.bitcoinj.base.exceptions.AddressFormatException;
+
+
+/**
+ * Interface for parsing and validating address strings.
+ * TODO: Move this class to o.b.base
+ */
+public interface AddressParser {
+    /**
+     * Parse an address that could be for any network
+     * @param addressString string representation of address
+     * @return A validated address object
+     * @throws AddressFormatException invalid address string
+     */
+    Address parseAddress(String addressString) throws AddressFormatException;
```

Comment:
Perhaps rename this to `parseAddressAnyNetwork`?

Reply:
Yes! I'll change it.

Reply:
Fixed, rebased, and force-pushed.